sky/cli.py

```diff
@@ -4865,24 +4873,41 @@ def serve_logs(
         \b
         # Tail the logs of replica 1
         sky serve logs [SERVICE_NAME] 1
+        \b
+        # Sync down logs for controller, load balancer, and all
+        # replicas for a service
+        sky serve logs [SERVICE_NAME] --sync-down
+        # Sync down logs for only controller. Same pattern
+        applies for load balancer
+        sky serve logs [SERVICE_NAME] --controller --sync-down
+        # Sync down logs only for a single replica
+        sky serve logs [SERVICE_NAME] 3 --sync-down
     """
     have_replica_id = replica_id is not None
     num_flags = (controller + load_balancer + have_replica_id)
     if num_flags > 1:
         raise click.UsageError('At most one of --controller, --load-balancer, '
-                               '[REPLICA_ID] can be specified.')
-    if num_flags == 0:
+                               '[REPLICA_ID], -s / --sync-down can be '
+                               'specified.')
+    if num_flags == 0 and not sync_down:
         raise click.UsageError('One of --controller, --load-balancer, '
-                               '[REPLICA_ID] must be specified.')
+                               '[REPLICA_ID] must be specified if not '
+                               'using --sync-down.')
+
     if controller:
         target_component = serve_lib.ServiceComponent.CONTROLLER
     elif load_balancer:
         target_component = serve_lib.ServiceComponent.LOAD_BALANCER
-    else:
-        # Already checked that num_flags == 1.
-        assert replica_id is not None
+    elif replica_id:
```

Comment:
```suggestion
    elif replica_id is not None:
```
nit: explicitly check None. ref https://google.github.io/styleguide/pyguide.html#2144-decision